crates/gen-js/src/lib.rs

```diff
@@ -189,7 +189,7 @@ impl Js {
                     TypeDefKind::Variant(v) => {
                         if self.is_nullable_option(iface, v) {
                             self.print_ty(iface, v.cases[1].ty.as_ref().unwrap());
-                            self.src.ts(" | null");
+                            self.src.ts(" | null")
```

Comment:
I think this is a stray change now?

Reply:
No you still need it for cases like function signatures.  While you can have:
```ts
function foo(x?: string);
```

It has the requirement that the optional param comes after any non-optional ones, which isn't the case generally in wit, e.g. would fail

```ts
function foo(x?: string, y: number); 
function foo(x: string | null, y: number);  // <--- forces caller to put null
```

Perhaps if wit makes the same requirement about option types we could remove this.

Reply:
I mean moreso the drop of the semicolon